core/vote/vote_journal.go

```diff
@@ -61,15 +91,16 @@ func (journal *VoteJournal) WriteVote(voteMessage *types.VoteEnvelope) error {
 
 	firstIndex, err := walLog.FirstIndex()
 	if err != nil {
-		log.Warn("Failed to get first index of votes journal", "err", err)
+		log.Error("Failed to get first index of votes journal", "err", err)
 	}
 
 	if lastIndex-firstIndex+1 > maxSizeOfRecentEntry {
 		if err := walLog.TruncateFront(lastIndex - maxSizeOfRecentEntry + 1); err != nil {
-			log.Warn("Failed to truncate votes journal", "err", err)
+			log.Error("Failed to truncate votes journal", "err", err)
 		}
 	}
 
+	journal.voteDataBuffer.Add(voteMessage.Data.TargetNumber, voteMessage.Data)
```

Comment:
if journal.voteDataBuffer.Len() > maxSizeOfRecentEntry {
		journal.voteDataBuffer.RemoveOldest()
	}

Reply:
I didn't get your mean. LRU has the eviction mechanism, right?